modules/nf-core/bascule/templates/main_script.R

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env Rscript
+
+parse_args = function(x) {
+  x = gsub("\\\\[","",x)
+  x = gsub("\\\\]","",x)
+  # giving errors when we have lists like c(xxx, xxx) since it will separate it
+  # args_list = unlist(strsplit(x, ', ')[[1]])
+  args_list = unlist(strsplit(x, ", (?=[^)]*(?:\\\\(|\$))", perl=TRUE))
+  # args_vals = lapply(args_list, function(x) strsplit(x, split=":")[[1]])
+  args_vals = lapply(args_list, function(x) {
+    x_splt = strsplit(x, split=":")[[1]]
+    c(x_splt[1],  paste(x_splt[2:length(x_splt)], collapse=":"))
+  })
+
+  # Ensure the option vectors are length 2 (key/ value) to catch empty ones
+  args_vals = lapply(args_vals, function(z){ length(z) = 2; z})
+
+  parsed_args = structure(lapply(args_vals, function(x) x[2]), names = lapply(args_vals, function(x) x[1]))
+  parsed_args[! is.na(parsed_args)]
+}
+
+opt = list(
```

Comment:
There is more than one argument used in the script, so please set reasonable defaults in opt. Also, are the parameters set by args? Is it possible to move some of them into the task definition, allowing users to change it without messing around in configs?
I'm not familiar with bascule - to me the question is how likely are users going to want to change any parameter.
If very likely - I think it should be in the nf file.
If very unlikely, it can stay only in args.

Reply:
If this is a situation where (some) parameters should be changed together, I think there are best practices for doing so.